plugins/gssapi.c

```diff
@@ -2113,17 +2113,66 @@ static int gssapi_client_mech_step(void *conn_context,
 
 	/* Setup req_flags properly */
 	req_flags = GSS_C_MUTUAL_FLAG | GSS_C_SEQUENCE_FLAG;
-	if (params->props.max_ssf > params->external_ssf) {
-	    /* We are requesting a security layer */
+        if (text->mech_type == NULL) {
+            /* in SASL/GSSAPI RFC4752 requires the INTEG flag MUST be set to
+             * TRUE in all cases */
```

Comment:
Because it is not set when maxssf = 0 or no security is requested at all.

That "all cases" means IIRC all cases when any protection is requested maxssf >= 1

Ie you are not allowed to request only Confidentiality, you must always request integrity when Confidentiality is requested.